### PR TITLE
CBL-2541 add a new PBKDF based on SHA1.

### PR DESCRIPTION
**Pull request by jianminzhao**
This function will be phased out after the entire gamut of CBL uses SHA256 based PBKDF.

### CODE REVIEW

**Review by borrrden — CHANGES_REQUESTED** [review]

File: C/include/c4Database.h [hunk 1]
```diff
@@ -36,16 +36,18 @@ C4API_BEGIN_DECLS
...
+           encryptionKey. As output, it will be the actual length required, or filled up, by the algorithm.
+           If the required length is greater the given length, this function returns false.
+           32 is a good start.
         @param password  The password string.
-        @param alg  The encryption algorithm to use. Must not be kC4EncryptionNone.
         @return  True on success, false on failure */
-    bool c4key_setPassword(C4EncryptionKey *encryptionKey,
-                           C4String password,
-                           C4EncryptionAlgorithm alg) C4API;
+    bool c4key_setPasswordSHA1(uint8_t* C4NONNULL encryptionKey,
```
[the remark this answers is not in the document]
Comment (pasin) [reply C/include/c4Database.h:48]: Should it be the same signature but include SHA1 in C4EncryptionAlgorithm?

File: C/Cpp_include/c4Database.hh [hunk 2]
```diff
@@ -37,6 +38,8 @@ C4_ASSUME_NONNULL_BEGIN
 C4EncryptionKey C4EncryptionKeyFromPassword(fleece::slice password,
                                             C4EncryptionAlgorithm = kC4EncryptionAES256);
 
+/// Derives an encryption key from a user-entered password with SHA1 as the hashing function
+std::array<uint8_t,32> C4EncryptionKeyFromPasswordSHA1(fleece::slice password);
```
Comment (pasin) [inline C/Cpp_include/c4Database.hh:42]: Should it be the same signature as C4EncryptionKeyFromPassword but include SHA1 in C4EncryptionAlgorithm?

File: C/include/c4Database.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (pasin) [reply C/include/c4Database.h:48]: Shouldn't we use the original signature but add SHA-1 to the algorithm enum?

[the remark this answers is not in the document]
Comment (borrrden) [reply C/include/c4Database.h:48]: No, the enum is concerning the *encryption* algorithm.  The hashing algorithm is a separate concern.

[the remark this answers is not in the document]
Comment (pasin) [reply C/include/c4Database.h:48]: Umm .. this seems like a new enum (e.g. C4PasswordDerivationAlgorithm) should be added and used but this is probably out of this PR's scope?

File: C/c4Database.cc [hunk 3]
```diff
@@ -54,6 +54,13 @@ C4EncryptionKey C4EncryptionKeyFromPassword(slice password, C4EncryptionAlgorith
     return key;
 }
 
+std::array<uint8_t,32> C4EncryptionKeyFromPasswordSHA1(fleece::slice password) {
+    std::array<uint8_t,32> key{};
+    AssertParam(password.size > 0, "Password is empty");
+    if (!litecore::DeriveKeyFromPasswordSHA1(password, key.data(), key.size()))
+        C4Error::raise(LiteCoreDomain, kC4ErrorCrypto, "SHA1 Key derivation failed");
```
Comment (pasin) [inline C/c4Database.cc:61]: In C4EncryptionKeyFromPassword, we also set algorithm to the key as well. Do we need to set anything here?

File: C/include/c4Database.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jianminzhao) [reply C/include/c4Database.h:48]: Okay, let me change it.

[the remark this answers is not in the document]
Comment (pasin) [reply C/include/c4Database.h:48]: I think It got it. Please ignore my comment.

**Comment by pasin** [comment]
Test Built Error

**Review by pasin — APPROVED** [review]

**Review by borrrden — APPROVED** [review]

**Pull request merged**